extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -206,11 +206,11 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	public getBooks(): BookTreeItem[] {
 		let books: BookTreeItem[] = [];
-		for (let i in this._tableOfContentPaths) {
-			let root = path.dirname(path.dirname(this._tableOfContentPaths[i]));
+		for (const tocPath of this._tableOfContentPaths) {
```

Comment:
Array#map would probably be better here, would automatically create the books array: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Possibly - but I don't know of an easy way to handle the logic this method has where exceptions caused while loading cause the item to be skipped.

I guess I could just do a map and have it return undefined if an error occurs, then filter out all undefined values. But at that point I'm not really sure we're gaining anything there from using map.

Reply:
I already did this in my PR https://github.com/microsoft/azuredatastudio/pull/7037

Reply:
I did not realize that you would also work on the books stuff. I have lot changes in that pr  and it's  been a week that I sent it out. We'll have merge conflicts to deal with once either of our PR's are in. More PR's more conflicts to resolve. 
I understand that we need have smaller PR's to avoid this and will keep in mind from now on. Can you please go through it and let me merge that in asap.